src/core/halide_utils.cc

```diff
@@ -110,6 +110,10 @@ std::vector<DLTensorUPtr> inferOutputTensorInfo(
       }
       sizes.push_back(static_cast<int>(*c));
     }
+    // Write into a 0-D tensor
+    if (sizes.size() == 0) {
+      sizes.push_back(1);
+    }
```

Comment:
It's not clear to me whether this is a fix or a workaround.
Why not do #261 instead?
That is, does dlpack itself support 0D tensors?

Reply:
I think #261 works because ATen supports 0-D tensors and treats them as 1-D tensor with a single element.
dlpack itself is just a minimal in-memory format and does not provide implementations for operations on tensors.
It is not clear to me that #261 would just work out of the box with caffe2 or whatever other tensor library we may hook up to in the future.